Pilot/MessageSender.py

```diff
@@ -0,0 +1,266 @@
+""" Message sender module for the remote loggin system.
+    It provides the general interface of the message sender and
+    several implementations that e.g. allows to send the message
+    to a REST interface or to a MQ server using the Stomp protocol.
+"""
+
+import Queue
+import logging
+import stomp
+import requests
+
+
+class MessageSender(object):
+  """ General interface of message sender.
+  """
+
+  def sendMessage(self, msg, flag):
+    """ Must be implemented by children classes.
+    """
+    raise NotImplementedError
+
+
+def createMessageSender(senderType, params):
+  """
+  Function creates MessageSender according to sender type.
+  Args:
+    senderType(str): sender type to be created.
+                    The allowed types are 'LOCAL_FILE', 'MQ', 'REST_API',
+    params(dict): additional parameters passed to init
+  Return:
+    MessageSender or None in case of errors
+
+  """
+
+  try:
+    if senderType == 'LOCAL_FILE':
+      return LocalFileSender(params)
+    elif senderType == 'MQ':
+      return StompSender(params)
+    elif senderType == 'REST_API':
+      return RESTSender(params)
+    else:
```

Comment:
This series of `if..else` is not very dynamic. What I would do is do a runtime loading, using `getattr` and `__import__` (there are several examples in the code). Like this, adding a new "Sender" would be easy. This of course would mean that the types would need to correspond to what is actually in the code, e.g. type "LocalFile" would load "LocalFileSender", type "StompSender" and so on.

EDIT: in fact, there's already in pilotTools the `ObjectLoader` module that can probably be used for this purpose, and would load also from the pilot extensions.